core/src/main/java/org/apache/carbondata/core/scan/expression/MatchExpression.java

```diff
@@ -29,15 +29,19 @@
 @InterfaceAudience.Internal
 public class MatchExpression extends Expression {
   private String queryString;
-  public static String maxDoc;
+  private String maxDoc;
+
+  public MatchExpression(String queryString) {
+    this.queryString = queryString;
+  }
 
   public MatchExpression(String queryString, String maxDoc) {
     this.queryString = queryString;
-    setMaxDoc(maxDoc);
+    this.maxDoc = maxDoc;
```

Comment:
validate the maxDoc parameter here and convert it to int

Reply:
ok